File: NETCoreMail/MailConfig.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Newtonsoft.Json

Public Class MailConfig

    Inherits MailCfg

    Private _FileConfig As String

    Public Sub New()

    End Sub

    Public Sub New(ByVal FileConfig As String)
        _FileConfig = FileConfig
        Dim Ruta As String = System.IO.Path.GetDirectoryName(_FileConfig)
        System.IO.Directory.CreateDirectory(Ruta)
        Read()
    End Sub

    Public Sub Save()
        Try
            Dim json = JsonConvert.SerializeObject(Me, Newtonsoft.Json.Formatting.Indented)
            System.IO.File.WriteAllText(_FileConfig, json)
        Catch ex As Exception
            Throw New Exception($"Error en NETCoreMail.MailConfig.Save : {ex.Message}")
        End Try
    End Sub

    Private Sub Read()
        Try

            If Not System.IO.File.Exists(_FileConfig) Then
                Me.FROM = "alertasbas@bas.com.ar"
                Me.SMTP_USERNAME = "alertasbas"
                Me.SMTP_PASSWORD = "nuncacaduca"
                Me.SSL = False
                Me.HOST = "mail.bas.com.ar"
                Me.PORT = 587
                Save()
            Else
                Dim json = System.IO.File.ReadAllText(_FileConfig)
                Dim Cfg As MailConfig = JsonConvert.DeserializeObject(Of MailConfig)(json)
                Me.FROM = Cfg.FROM
                Me.SMTP_USERNAME = Cfg.SMTP_USERNAME
                Me.SMTP_PASSWORD = Cfg.SMTP_PASSWORD
                Me.HOST = Cfg.HOST
                Me.PORT = Cfg.PORT
            End If

        Catch ex As Exception
            Throw New Exception($"Error en NETCoreMail.MailConfig.Read : {ex.Message}")
        End Try
    End Sub
End Class

Public Class MailCfg
    Public Sub New()

    End Sub

    Public Sub New(ByVal FROM As String, ByVal SMTP_USERNAME As String, ByVal SMTP_PASSWORD As String, ByVal SSL As Boolean, ByVal HOST As String, ByVal PORT As Integer)
        Me.FROM = FROM
        Me.SMTP_USERNAME = SMTP_USERNAME
        Me.SMTP_PASSWORD = SMTP_PASSWORD
        Me.SSL = SSL
        Me.HOST = HOST
        Me.PORT = PORT
    End Sub

    Public FROM As String = String.Empty
    Public SMTP_USERNAME As String = String.Empty
    Public SMTP_PASSWORD As String = String.Empty
    Public SSL As Boolean = False
    Public HOST As String = String.Empty
    Public PORT As Integer = 0
End Class

File: NETCoreMail_C/NETCoreMail/MailConfig.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Newtonsoft.Json;

namespace NETCoreMail
{
    class MailConfig : MailCfg
    {
       
        string _FileConfig;

        public MailConfig()
        {
            string Ruta = System.IO.Path.GetDirectoryName(_FileConfig);
            System.IO.Directory.CreateDirectory(Ruta);
        }

        public MailConfig(string FileConfig)
        {
            _FileConfig = FileConfig;
            Read();
        }
                
        public void Save()
        {
            try
            {
                var json = JsonConvert.SerializeObject(this, Newtonsoft.Json.Formatting.Indented);
                System.IO.File.WriteAllText(_FileConfig, json);
            }
            catch (Exception ex)
            {
              throw new Exception ($"Error en NETCoreMail.MailConfig.Save : {ex.Message}");
            }
        }

        private void Read()
        {
            try
            {

                if (!System.IO.File.Exists(_FileConfig))
                {
                    this.FROM = "alertasbas@bas.com.ar";
                    this.SMTP_USERNAME = "alertasbas";
                    this.SMTP_PASSWORD = "nuncacaduca";
                    this.SSL  = false;
                    this.HOST = "mail.bas.com.ar";
                    this.PORT = 587;
                    Save();
                    
                }

                else
                {
                    var json = System.IO.File.ReadAllText(_FileConfig);
                    MailConfig Cfg = JsonConvert.DeserializeObject<MailConfig>(json);
                    this.FROM = Cfg.FROM;
                    this.SMTP_USERNAME = Cfg.SMTP_USERNAME;
                    this.SMTP_PASSWORD = Cfg.SMTP_PASSWORD;
                    this.HOST = Cfg.HOST;
                    this.PORT = Cfg.PORT;
                }
            }
            catch (Exception ex)
            {
                throw new Exception($"Error en NETCoreMail.MailConfig.Read : {ex.Message}");
            }
        }
    }

    class MailCfg
    {
        public  MailCfg()
        {

        }

        public MailCfg(string FROM,
                       string SMTP_USERNAME,
                       string SMTP_PASSWORD,
                       bool SSL,
                       string HOST,
                       int PORT)
        {
            this.FROM = FROM;
            this.SMTP_USERNAME = SMTP_USERNAME;
            this.SMTP_PASSWORD = SMTP_PASSWORD;
            this.SSL = SSL;
            this.HOST = HOST;
            this.PORT = PORT;
        }

        public string FROM = string.Empty;
        public string SMTP_USERNAME = string.Empty;
        public string SMTP_PASSWORD = string.Empty;
        public bool SSL = false;
        public string HOST = string.Empty;
        public int PORT = 0;
    }
}
